ccs-scale-infra-shared/terraform/modules/infrastructure/cloudfront/functions/security-headers/index.js

```diff
@@ -12,7 +12,7 @@ exports.handler = (event, context, callback) => {
   }];
   headers['content-security-policy'] = [{
     key: 'Content-Security-Policy',
-    value: "default-src 'none'; img-src 'self'; script-src 'self' 'unsafe-inline'; font-src fonts.gstatic.com; style-src 'self' 'unsafe-inline' fonts.googleapis.com; object-src 'none'"
+    value: "default-src 'none'; img-src 'self'; script-src 'self' 'nonce-{SERVER-GENERATED-NONCE}'; img-src www.googletagmanager.com 'unsafe-inline'; font-src fonts.gstatic.com; style-src 'self' 'unsafe-inline' fonts.googleapis.com; object-src 'none'"
```

Comment:
From the google documentation - looks like the suggested is `Content-Security-Policy: script-src 'nonce-{SERVER-GENERATED-NONCE}'; img-src www.googletagmanager.com`.

There is an existing `img-src` directive which `www.googletagmanager.com` can be added to rather than creating a new one (as this has now split off the `'unsafe-inline'` from the `script-src` directive - maybe something like below?

```
value: "default-src 'none'; img-src 'self' www.googletagmanager.com; script-src 'self' 'unsafe-inline' 'nonce-{SERVER-GENERATED-NONCE}'; font-src fonts.gstatic.com; style-src 'self' 'unsafe-inline' fonts.googleapis.com; object-src 'none'"
```

Reply:
Ali has been  looking into this further and has the following analysis.
I don't think this solution will work for 2 reasons: 
1. the server generated nonce will have to be generated on the server configuration, so the file you just sent, but then this nonce needs to also be sent as a variable to the ccs-scale-ui frontend to be used in the code there I'm not sure how you would do this or wether this would even be possible? 

2. Nonce is not supported in Internet explorer and so data won’t be sent from this browser. 

I think the first point is probably the most problematic. 
The value should be something like this 
`value: "default-src 'none'; img-src 'self' www.googletagmanager.com; script-src 'self' 'unsafe-inline'; font-src fonts.gstatic.com; style-src 'self' 'unsafe-inline' fonts.googleapis.com https://tagmanager.google.com/ https://www.googletagmanager.com/; ‘unsafe-eval' 'unsafe-inline' https://tagmanager.google.com/ https://www.googletagmanager.com/; object-src 'none'"`

Reply:
According to this documentation - https://developers.google.com/tag-manager/web/csp - it says that if using a nonce is not possible - you just need to add the following:

```
script-src: 'unsafe-inline' https://www.googletagmanager.com
img-src: www.googletagmanager.com
```

Which would just add to the the `script-src` and `img-src` directives, giving something more like:

```
value: "default-src 'none'; img-src 'self' www.googletagmanager.com; script-src 'self' 'unsafe-inline' https://www.googletagmanager.com; font-src fonts.gstatic.com; style-src 'self' 'unsafe-inline' fonts.googleapis.com; object-src 'none'"
```

In the PR it is not adding to `script-src` but is adding to the `style-src` (which also looks to have a semi-colon not followed by a directive). Is this intentional?

Reply:
The one in the Pr is outdated and I have just updated it to the suggested value.
There's a caveat in the documentation .The use of 'unsafe-inline' is discouraged, and one must carefully consider the security ramifications of adding this directive to the CSP before using this approach.
Some more documentation available here.
https://content-security-policy.com/
https://content-security-policy.com/browser-test/

Reply:
👍  the `unsafe-inline` was probably added for a reason during the development and was approved via the pen test. If it needs to be reviewed should probably be done as a separate task and regression tested.